packages/stream-model-handler/src/account-relations-utils.ts

```diff
@@ -0,0 +1,48 @@
+import { JsonReference } from 'json-ptr'
+import type { JSONSchema } from 'json-schema-typed/draft-2020-12'
+
+type FieldSchema = Exclude<JSONSchema, boolean>
+
+const SUPPORTED_FIELD_TYPES = ['boolean', 'integer', 'number', 'string']
+
+export function validateSetFields(fields: Array<string>, modelSchema: JSONSchema.Object): void {
+  if (fields.length === 0) {
+    throw new Error(`At least one field must be defined for the SET account relation`)
+  }
+
+  const requiredFields = modelSchema.required ?? []
+
+  const errors: Array<Error> = []
+  for (const field of fields) {
+    let fieldSchema = modelSchema.properties[field] as FieldSchema | undefined
+    if (fieldSchema?.$ref != null) {
+      const ref = new JsonReference(fieldSchema.$ref)
```

Comment:
I don't really understand what this block is doing?  Can you maybe add some comments here?

Reply:
I still don't entirely understand, but I'm just going to chalk this up to some jsonSchema shenanigans and move on 😅

Reply:
Here is an example schema generated by the ComposeDB tools:

```json
{
  "$defs": {
    "ImageMetadata": {
      "additionalProperties": false,
      "properties": {
        "height": {
          "minimum": 1,
          "type": "integer",
        },
        "mimeType": {
          "maxLength": 50,
          "type": "string",
        },
        "size": {
          "minimum": 1,
          "type": "integer",
        },
        "src": {
          "maxLength": 150,
          "type": "string",
        },
        "width": {
          "minimum": 1,
          "type": "integer",
        },
      },
      "required": [
        "src",
        "mimeType",
        "width",
        "height",
      ],
      "title": "ImageMetadata",
      "type": "object",
    },
    "ImageSources": {
      "additionalProperties": false,
      "properties": {
        "alternatives": {
          "items": {
            "$ref": "#/$defs/ImageMetadata",
          },
          "maxItems": 20,
          "type": "array",
        },
        "original": {
          "$ref": "#/$defs/ImageMetadata",
        },
      },
      "required": [
        "original",
      ],
      "title": "ImageSources",
      "type": "object",
    },
  },
  "$schema": "https://json-schema.org/draft/2020-12/schema",
  "additionalProperties": false,
  "properties": {
    "background": {
      "$ref": "#/$defs/ImageSources",
    },
    "description": {
      "maxLength": 420,
      "type": "string",
    },
    "emoji": {
      "maxLength": 2,
      "type": "string",
    },
    "url": {
      "maxLength": 240,
      "type": "string",
    },
  },
  "type": "object",
}
```

In the `$defs` object we can have shared types that can be used (referenced) by other types in the schema, for example `ImageSources` uses `ImageMetadata` in 2 places, and the `background` property of the schema uses `ImageSources`.
This notably allows to avoid duplication in the JSON schema and therefore also in the runtime GraphQL schema.

So here regarding the logic itself, we use `JsonReference` to resolve the reference in order to get the actual structure of the field so we can check its type.